src/main/java/org/folio/services/ftp/SftpUploadService.java

```diff
@@ -0,0 +1,121 @@
+    factory.setHost(server);
+    factory.setPort(port);
+    factory.setUsername(username);
+    factory.setPassword(password);
+    factory.setSshClient(ssh);
+    factory.setConnectTimeout(TimeUnit.SECONDS.toMillis(30L));
+    factory.setAuthenticationTimeout(TimeUnit.SECONDS.toMillis(30L));
+    factory.afterPropertiesSet();
+    return factory;
+  }
+
+  public Future<String> upload(Context ctx, String username, String password, String folder, String filename, String content)
+      throws Exception {
+    Promise<String> promise = Promise.promise();
+    String folderPath = StringUtils.isEmpty(folder) ? "" : (folder + FILE_SEPARATOR);
+    String remoteAbsPath = folderPath + filename;
+
+    SessionFactory<SftpClient.DirEntry> sshdFactory;
+    try {
+      sshdFactory = getSshdSessionFactory(username, password);
+    } catch (Exception e) {
+      throw new FtpException(HttpStatus.HTTP_FORBIDDEN.toInt(),String.format("Unable to connect to %s:%d", server, port));
+    }
+    ctx.owner().executeBlocking(blockingFeature -> {
+      try (InputStream inputStream = new ByteArrayInputStream(content.getBytes()); var session = sshdFactory.getSession()) {
+        logger.info("Start uploading file to SFTP path: {}", remoteAbsPath);
+
+        createRemoteDirectoryIfAbsent(session, folder);
+        session.write(inputStream, remoteAbsPath);
+        logger.info("uploaded: {}", remoteAbsPath);
+        blockingFeature.complete("Uploaded successfully");
+      } catch (Exception e) {
+        logger.info("Error uploading the file {}", remoteAbsPath, e);
+        blockingFeature.fail(new CompletionException(e));
+      } finally {
+        logger.info("Session closed");
+        sshdFactory.getSession().close();
+      }
+    }, false, asyncResultHandler(promise));
+    return promise.future();
+  }
+
+  private void createRemoteDirectoryIfAbsent(Session<SftpClient.DirEntry> session, String folder) throws IOException {
+    if (!session.exists(folder)) {
+      String[] folders = folder.split("/");
+      StringBuilder path = new StringBuilder(folders[0]).append("/");
+
+      for (int i = 0; i < folders.length; i++) {
+        if (!session.exists(path.toString())) {
+          session.mkdir(path.toString());
+        }
+        if (i == folders.length - 1) return;
+        path.append(folders[i + 1]).append("/");
+      }
+      logger.info("A directory has been created: {}", folder);
+    }
+  }
+
+  private Handler<AsyncResult<String>> asyncResultHandler(Promise<String> promise) {
+    return result -> {
+      if (result.succeeded()) {
+        logger.debug("Success upload to SFTP");
+        promise.complete(result.result());
+      } else {
+        logger.error("Failed upload to SFTP");
+        String message = Optional.ofNullable(result.cause())
+          .map(Throwable::getMessage)
+          .orElse(result.cause().getMessage());
+        logger.error(message);
```

Comment:
```suggestion
        logger.error("Failed upload to SFTP", result.cause());
```
and this code can be deleted:
`logger.error("Failed upload to SFTP");
        String message = Optional.ofNullable(result.cause())
          .map(Throwable::getMessage)
          .orElse(result.cause().getMessage());`

Reply:
fixed